### PR TITLE
ocm:server-foundation add sonarcloud

### PR DESCRIPTION
**Pull request by zhiweiyin318**

### CODE REVIEW

File: ci-operator/jobs/open-cluster-management/multicloud-operators-foundation/open-cluster-management-multicloud-operators-foundation-main-presubmits.yaml [hunk 1]
```diff
@@ -306,3 +361,4 @@ presubmits:
         secret:
           secretName: result-aggregator
     trigger: (?m)^/test( | .* )verify-deps,?($|\s.*)
+
```
Comment (qiujian16) [inline ci-operator/jobs/open-cluster-management/multicloud-operators-foundation/open-cluster-management-multicloud-operators-foundation-main-presubmits.yaml:364]: eof?

Comment (zhiweiyin318) [reply ci-operator/jobs/open-cluster-management/multicloud-operators-foundation/open-cluster-management-multicloud-operators-foundation-main-presubmits.yaml]: fixed

File: ci-operator/config/open-cluster-management/multicloud-operators-foundation/open-cluster-management-multicloud-operators-foundation-main.yaml [hunk 2]
```diff
@@ -3,6 +3,10 @@ base_images:
     name: ubi-minimal
     namespace: ocp
     tag: "8"
+  open-cluster-management_builder_go1.15-linux-amd64:
```
Comment (ray-harris) [inline ci-operator/config/open-cluster-management/multicloud-operators-foundation/open-cluster-management-multicloud-operators-foundation-main.yaml:6]: You don't need this part. Base images define external images to use as the "from:" field in the "images:" stanza. For "images:" you only need "base".

File: ci-operator/config/open-cluster-management/multicloud-operators-foundation/open-cluster-management-multicloud-operators-foundation-main.yaml [hunk 3]
```diff
@@ -64,6 +68,15 @@ tests:
         requests:
           cpu: 100m
     workflow: ipi-aws
+- as: sonarcloud
+  commands: |
+    export SELF="make -f /opt/build-harness/Makefile.prow"
+    make -f /opt/build-harness/Makefile.prow sonar/go/prow
+  container:
+    from: open-cluster-management_builder_go1.15-linux-amd64
```
Comment (ray-harris) [inline ci-operator/config/open-cluster-management/multicloud-operators-foundation/open-cluster-management-multicloud-operators-foundation-main.yaml:76]: This needs to be "from: src". The "open-cluster-management_builder_go1.15-linux-amd64" doesn't have your source code injected into it.

The "src" image is automatically built by Prow by taking the "build_root" image and injecting your source code into it.

**Review by ray-harris — CHANGES_REQUESTED** [review]

**Pull request merged**